programs/bpf_loader/src/syscalls/cpi.rs

```diff
@@ -28,43 +80,61 @@ struct CallerAccount<'a> {
     // mapped inside the vm (see serialize_parameters() in
     // BpfExecutor::execute).
     //
-    // When direct mapping is off, this includes both the account data _and_ the
-    // realloc padding. When direct mapping is on, account data is mapped in its
-    // own separate memory region which is directly mutated from inside the vm,
-    // and the serialized buffer includes only the realloc padding.
+    // This is only set when direct mapping is off (see the relevant comment in
+    // CallerAccount::from_account_info).
     serialized_data: &'a mut [u8],
     // Given the corresponding input AccountInfo::data, vm_data_addr points to
     // the pointer field and ref_to_len_in_vm points to the length field.
     vm_data_addr: u64,
-    ref_to_len_in_vm: &'a mut u64,
+    ref_to_len_in_vm: VmValue<'b, 'a, u64>,
```

Comment:
nit: seems lifetime params is swapped for no good reason? maybe no harm, but this code is passing a lifetime called `'b` as `VmValue::'a`

hope clippy had equivalent of this: https://rust-lang.github.io/rust-clippy/master/index.html#/mismatching_type_param_order

Reply:
the reason is `'a` is the same lifetime in `CallerAccount` and `VmValue`: the lifetime of the _content_ of a memory mapping. `'b` is the lifetime of the `MemoryMapping` reference itself.

If I swap them, then you have

```
struct CallerAccount<'a, 'b> {
    lamports: &'a mut u64,
    ...
```

and 

```
enum VmValue<'a, 'b, T> {
    ...
    Translated(&'b mut T),
```

And I find this _more_ confusing since `'a` and `'b` are actually the same lifetime